packages/core/src/html/NodeProcessor.js

```diff
@@ -597,141 +140,85 @@ class NodeProcessor {
     try {
-      NodeProcessor.transformOldSlotSyntax(node);
-      NodeProcessor.shiftSlotNodeDeeper(node);
+      transformOldSlotSyntax(node);
+      shiftSlotNodeDeeper(node);
+
+      // log warnings for deprecated and conflicting attributes
+      if (_.has(warnDeprecatedAtributesMap, node.name)) { warnDeprecatedAtributesMap[node.name](node); }
+      if (_.has(warnConflictingAtributesMap, node.name)) { warnConflictingAtributesMap[node.name](node); }
 
       switch (node.name) {
       case 'frontmatter':
         this._processFrontMatter(node, context);
         break;
       case 'body':
-        NodeProcessor._preprocessBody(node);
+        console.warn(`<body> tag found in ${node.attribs[ATTRIB_CWF]}. This may cause formatting errors.`);
         break;
       case 'code':
         node.attribs['v-pre'] = '';
         break;
       case 'include':
-        this.docId += 1; // used in markdown-it-footnotes
+        this.markdownProcessor.docId += 1; // used in markdown-it-footnotes
         return processInclude(node, context, this.pageSources, this.variableProcessor,
-                              text => this._renderMd(text), text => this._renderMdInline(text),
+                              text => this.markdownProcessor.renderMd(text),
+                              text => this.markdownProcessor.renderMdInline(text),
                               this.config);
       case 'panel':
-        this._processPanelAttributes(node);
+        this.componentProcessor.processPanelAttributes(node);
         return processPanelSrc(node, context, this.pageSources, this.config);
       case 'question':
-        this._processQuestion(node);
+        this.componentProcessor.processQuestion(node);
         break;
       case 'q-option':
-        this._processQOption(node);
+        this.componentProcessor.processQOption(node);
         break;
       case 'quiz':
-        this._processQuiz(node);
+        this.componentProcessor.processQuiz(node);
         break;
       case 'popover':
-        this._processPopover(node);
+        this.componentProcessor.processPopover(node);
+        transformBootstrapVuePopover(node);
         break;
       case 'tooltip':
-        this._processTooltip(node);
+        this.componentProcessor.processTooltip(node);
+        transformBootstrapVueTooltip(node);
         break;
       case 'modal':
-        this._processModalAttributes(node);
+        this.componentProcessor.processModalAttributes(node);
+        transformBootstrapVueModalAttributes(node,
+                                             (originalName, newName) =>
+                                               NodeProcessor._renameSlot(node, originalName, newName),
+                                             (originalAttribute, newAttribute) =>
+                                             // eslint-disable-next-line max-len
+                                               NodeProcessor._renameAttribute(node, originalAttribute, newAttribute));
```

Comment:
could we move `_renameSlot` +`_renameAttribute` into `bootstrapvueprocessor`? since its only used there

Reply:
Okay I have moved the functions over